bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/ColumnDescriptorAdapter.java

```diff
@@ -148,33 +153,41 @@ public static void throwIfRequestingUnsupportedFeatures(HColumnDescriptor column
   /**
    * Construct an Bigtable GC expression from the given column descriptor.
    */
-  public static String buildGarbageCollectionExpression(HColumnDescriptor columnDescriptor) {
+  public static GcRule buildGarbageCollectionExpression(HColumnDescriptor columnDescriptor) {
     int maxVersions = columnDescriptor.getMaxVersions();
     int minVersions = columnDescriptor.getMinVersions();
     int ttlSeconds = columnDescriptor.getTimeToLive();
-    long bigtableTtl = BigtableConstants.BIGTABLE_TIMEUNIT.convert(ttlSeconds, TimeUnit.SECONDS);
 
     Preconditions.checkState(minVersions < maxVersions,
         "HColumnDescriptor min versions must be less than max versions.");
 
-    StringBuilder buffer = new StringBuilder();
-    if (ttlSeconds != HColumnDescriptor.DEFAULT_TTL) {
-      // minVersions only comes into play with a TTL:
-      if (minVersions != HColumnDescriptor.DEFAULT_MIN_VERSIONS) {
-        buffer.append(String.format("(age() > %s && version() > %s)", bigtableTtl, minVersions));
+    if (ttlSeconds == HColumnDescriptor.DEFAULT_TTL) {
+      if (maxVersions == Integer.MAX_VALUE) {
+        return null;
       } else {
-        buffer.append(String.format("(age() > %s)", bigtableTtl));
+        return createMaxVersionsRule(maxVersions);
       }
     }
-    // Since the HBase version default is 1, which may or may not be the same as Bigtable
-    if (buffer.length() != 0) {
-      buffer.append(" || ");
+
+    // minVersions only comes into play with a TTL:
+    GcRule ageRule =
+        GcRule.newBuilder().setMaxAge(Duration.newBuilder().setSeconds(ttlSeconds)).build();
+    if (minVersions != HColumnDescriptor.DEFAULT_MIN_VERSIONS) {
+      GcRule minVersionRule = createMaxVersionsRule(minVersions);
```

Comment:
This reads slightly strangely and looks like a typo at first glance so a comment  might be useful